### PR TITLE
SOC-2222 | Added discussion stats to wikiStats

### PR DESCRIPTION
**Pull request by Bloodmoon24**
## Links
- https://wikia-inc.atlassian.net/browse/SOC-2222
- https://github.com/Wikia/mercury/pull/2385
## Reviewers

@kvas-damian  @bkoval please CR

### CODE REVIEW

File: includes/wikia/services/WikiDetailsService.class.php [hunk 1]
```diff
@@ -8,6 +8,7 @@ class WikiDetailsService extends WikiService {
 	const DEFAULT_WIDTH = 250;
 	const DEFAULT_HEIGHT = null;
 	const DEFAULT_SNIPPET_LENGTH = null;
+	const DISCUSSION_ENDPOINT = 'https://services.wikia.com/discussion/';
```
Comment (rogatty) [inline includes/wikia/services/WikiDetailsService.class.php]: Shouldn't this go through consul?

File: includes/wikia/services/WikiDetailsService.class.php [hunk 2]
```diff
@@ -261,6 +263,11 @@ protected function cacheWikiData( $wikiInfo, $method = null ) {
 		$wgMemc->set( $key, $wikiInfo, static::CACHE_1_DAY );
 	}
 
+	private function getDiscussionStats( $id ) {
+		$response = json_decode(file_get_contents(self::DISCUSSION_ENDPOINT."$id/forums/$id"), true);
```
Comment (kvas-damian) [inline includes/wikia/services/WikiDetailsService.class.php]: Spaces, spaces, space: https://www.mediawiki.org/wiki/Manual:Coding_conventions/PHP#Spaces
What about error handling?

Comment (kvas-damian) [inline includes/wikia/services/WikiDetailsService.class.php]: What do you think about using some library for sending HTTP requests?

**Comment by kvas-damian** [comment]
What should happen if discussions are not enabled?

**Comment by Bloodmoon24** [comment]
getDiscussionStats() returns null, which is converted to 0. It's not displayed on frontend.

File: includes/wikia/services/WikiDetailsService.class.php (hunk 2 above)
Comment (rogatty) [inline includes/wikia/services/WikiDetailsService.class.php]: Check out https://github.com/Wikia/app/blob/733df9f5d176ef07e614828e30c2e1d2805371a0/extensions/wikia/Recirculation/services/DiscussionsDataService.class.php

**Comment by kvas-damian** [comment]
I believe we shouldn't include that data when discussions are disabled.

File: includes/wikia/services/WikiDetailsService.class.php [hunk 3]
```diff
@@ -262,6 +263,26 @@ protected function cacheWikiData( $wikiInfo, $method = null ) {
...
+	private function getDiscussionStats( $id ) {
+		global $wgConsulServiceTag, $wgConsulUrl, $wgEnableDiscussions;
+
+		if ( $wgEnableDiscussions ) {
+			$consulUrl = ( new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag ))->getUrl( 'discussion' );
+			$response = Http::get( "http://$consulUrl/$id/forums/$id", 'default', array( 'noProxy' => true ));
+			if ( $response != false ) {
+				$decodedResponse = json_decode( $response, true );
+				if ( isset($decodedResponse) && json_last_error() === JSON_ERROR_NONE ) {
+					return $decodedResponse[ 'threadCount' ] ? $decodedResponse[ 'threadCount' ] : null;
```
Comment (kvas-damian) [inline includes/wikia/services/WikiDetailsService.class.php]: I'm not sure this condition is valid.
When `$decodedResponse[ 'threadCount' ]` is `0` then we return `null`?

File: includes/wikia/services/WikiDetailsService.class.php [hunk 4]
```diff
@@ -262,6 +263,26 @@ protected function cacheWikiData( $wikiInfo, $method = null ) {
...
+	 */
+	private function getDiscussionStats( $id ) {
+		global $wgConsulServiceTag, $wgConsulUrl, $wgEnableDiscussions;
+
+		if ( $wgEnableDiscussions ) {
+			$consulUrl = ( new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag ))->getUrl( 'discussion' );
+			$response = Http::get( "http://$consulUrl/$id/forums/$id", 'default', array( 'noProxy' => true ));
+			if ( $response != false ) {
+				$decodedResponse = json_decode( $response, true );
+				if ( isset($decodedResponse) && json_last_error() === JSON_ERROR_NONE ) {
```
Comment (kvas-damian) [inline includes/wikia/services/WikiDetailsService.class.php]: spaces

**Comment by kvas-damian** [comment]
LGTM

**Comment by rafalkalinski** [comment]
LGTM

File: includes/wikia/services/WikiDetailsService.class.php [hunk 5]
```diff
@@ -190,7 +192,7 @@ protected function getFromService( $id ) {
 			}
 		}
 
-		return [
+		$stats = [
 			'stats' => [
```
Comment (rogatty) [inline includes/wikia/services/WikiDetailsService.class.php]: Fields like `headline` or `lang` are not stats ;) `$wikiDetails` would be more appropriate.

File: includes/wikia/services/WikiDetailsService.class.php [hunk 6]
```diff
@@ -190,7 +192,7 @@ protected function getFromService( $id ) {
 			}
 		}
 
-		return [
+		$stats = [
```
Comment (rogatty) [inline includes/wikia/services/WikiDetailsService.class.php]: Fields like `headline` or `lang` are not stats ;) `$wikiDetails` would be more appropriate.

**Pull request merged**